sdks/java/testing/load-tests/src/main/java/org/apache/beam/sdk/loadtests/SyntheticDataPubSubPublisher.java

```diff
@@ -83,14 +97,41 @@ public static void main(String[] args) throws IOException {
 
     Pipeline pipeline = Pipeline.create(options);
 
-    pipeline
-        .apply("Read synthetic data", Read.from(new SyntheticBoundedSource(sourceOptions)))
-        .apply("Map to PubSub messages", MapElements.via(new MapBytesToPubSubMessage()))
-        .apply("Write to PubSub", PubsubIO.writeMessages().to(options.getInsertionPipelineTopic()));
-
+    if (!options.getKafkaBootstrapServerAddress().isEmpty()) {
```

Comment:
Those sinks are somewhat independent and we could let them stay this way. We can code it like this (including the division of the methods):

```
if (<options for kafka are present) {
      writeToKafka(collection);

if (<options for pubsub are present>)
      writeToPubsub(collection);
}
```

Reply:
implemented, thanks for the suggestion!